ledger/participant-integration-api/src/main/scala/platform/store/interning/StringInterningView.scala

```diff
@@ -94,10 +104,17 @@ class StringInterningView(loadPrefixedEntries: LoadStringInterningEntries)
 
   override def update(
       lastStringInterningId: Int
-  )(implicit loggingContext: LoggingContext): Future[Unit] =
-    if (lastStringInterningId <= raw.lastId) {
-      Future.unit
-    } else {
+  )(implicit loggingContext: LoggingContext, executionContext: ExecutionContext): Future[Unit] =
+    if (lastStringInterningId <= raw.lastId) Future {
+      val idsToBeRemoved = (lastStringInterningId + 1 to raw.lastId).toSet
+      val stringsToBeRemoved = idsToBeRemoved.map(raw.idMap)
+
+      raw = raw.copy(
+        map = raw.map.removedAll(stringsToBeRemoved),
+        idMap = raw.idMap.removedAll(idsToBeRemoved),
+      )
+    }
+    else {
       loadPrefixedEntries(raw.lastId, lastStringInterningId)(loggingContext)
         .map(updateView)(ExecutionContext.parasitic)
```

Comment:
if we want to have an EC, then it should probably be used here

Reply:
Discussed offline. Keeping it as it is.